File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.js
function isPrime(num) {
    for(let i = 2; i < num; i++) {
        if(num % i === 0) return false;
    }
    return true;
}

function solution(nums) {
    let cnt = 0;
    
    for(let i = 0; i < nums.length; i++) {
        for(let j = i + 1; j < nums.length; j++) {
            for(let k = j + 1; k < nums.length; k++) {
                if(isPrime(nums[i] + nums[j] + nums[k])) cnt++; 
            }
        }
    }
    return cnt;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
// 더할 거니까 순서가 중요하지 않음, 중복 불가능 -> 조합
class Solution {
    static boolean[] visited;
    static int answer = 0;
    
    static boolean isPrime(int num) {
        for(int i = 2; i < num; i++) {
            if(num % i == 0) return false;
        }
        return true;
    }
    
    static void dfs(int depth, int r, int[] nums, int index, int sum) {
        if(depth == r) {
            if(isPrime(sum)) answer++;
            return;
        }
        
        // index부터
        for(int i = index; i < nums.length; i++) {
            if(!visited[i]) {
                visited[i] = true;
                dfs(depth + 1, r, nums, i + 1, sum + nums[i]);
                visited[i] = false;
            }
        }
        
    }
    
    public int solution(int[] nums) {
        visited = new boolean[nums.length];
        
        dfs(0, 3, nums, 0, 0);
        
        return answer;
    }
}